### PR TITLE
[systemtest][mm] Add test for scaling mm to zero

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

### Type of change

- New test

### Description

After #3412 we are able to deploy (scale) the KafkaMirrorMaker to zero replicas and the resource will be `Ready`. This PR tests this feature. 

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=KafkaMirrorMakerST#testScaleMirrorMakerToZero

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=MirrorMakerST#testScaleMirrorMakerToZero

File: systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMakerST.java [hunk 1]
```diff
@@ -663,6 +666,36 @@ void testScaleMirrorMakerSubresource() {
...
+            .done();
+
+        String mmDepName = KafkaMirrorMakerResources.deploymentName(CLUSTER_NAME);
+        List<String> mmPods = kubeClient().listPodNames("type", "kafka-mirror-maker");
+        assertThat(mmPods.size(), is(3));
+
+        LOGGER.info("Scaling MirrorMaker to zero");
+        KafkaMirrorMakerResource.replaceMirrorMakerResource(CLUSTER_NAME, mm -> mm.getSpec().setReplicas(0));
+
+        KafkaMirrorMakerUtils.waitForKafkaMirrorMakerReady(CLUSTER_NAME);
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMakerST.java:690]: How does this ensure that the previous status is not caught by this? I do not see any checks for observed generations etc.

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMakerST.java:690]: Gonna add it, thanks

File: systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMakerST.java [hunk 2]
```diff
@@ -663,6 +666,36 @@ void testScaleMirrorMakerSubresource() {
...
+        PodUtils.waitForPodsReady(kubeClient().getDeploymentSelectors(mmDepName), 0, true);
+
+        mmPods = kubeClient().listPodNames("type", "kafka-mirror-maker");
+        KafkaMirrorMakerStatus mmStatus = KafkaMirrorMakerResource.kafkaMirrorMakerClient().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getStatus();
+
+        assertThat(mmPods.size(), is(0));
+        assertThat(mmStatus.getConditions().get(0).getType(), is("Ready"));
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMakerST.java:698]: We maybe should take into consideration to create some Enum on CR type statuses. Would be better instead of hardcoding these values

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMakerST.java:699]: I have it on the list, as many things, so I will do it in another PR as it's not related just to this test

File: systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMakerST.java (hunk 1 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMakerST.java:691]: https://github.com/strimzi/strimzi-kafka-operator/pull/3438/files#diff-22f8df9b640e22843eae57444b9be687R700

Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMakerST.java:691]: Check is added

Comment (scholzj) [reply systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMakerST.java:691]: So should this line be removed now?

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=MirrorMakerST#testScaleMirrorMakerToZero

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=MirrorMakerST#testScaleMirrorMakerToZero

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=MirrorMakerST#testScaleMirrorMakerToZero

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=MirrorMakerST#testScaleMirrorMakerToZero
